### PR TITLE
Make progress on projection-matrix-handling frame functions

### PR DESCRIPTION
**Pull request by cadmic**
not fully matched but maybe good enough to fix the matrices in gz

### CODE REVIEW

**Review by Yanis42 — COMMENTED** [review]
👀 (if you wanna look at mm's dwarf I have it [there](https://github.com/Yanis42/oot-gc/tree/mm-j/debug/mm-j))

File: src/emulator/frame.c [hunk 1]
```diff
@@ -749,11 +752,155 @@ static bool frameLoadTile(Frame* pFrame, FrameTexture** ppTexture, s32 iTileCode
...
+        pFrame->eTypeProjection = FMP_ORTHOGRAPHIC;
+    }
+}
 
 // fn_8004A69C
 
-// fn_8004A89C
+bool fn_8004A89C(Frame* pFrame) {
+    GXColor color;
+    s32 nFogType;
```
[the remark this answers is not in the document]
Comment (Yanis42) [reply src/emulator/frame.c:846]: I'm thinking this function could be either `frameDrawSetupFog_Zelda1` or `frameDrawSetupFog_Default`, based on the usage

[the remark this answers is not in the document]
Comment (cadmic) [reply src/emulator/frame.c]: `GXFogType` works yeah. I went with `frameDrawSetupFog_Default`, maybe there originally was `frameDrawSetupFog_Zelda1` too but they switch to the default version later for some reason

**Review by Yanis42 — APPROVED** [review]
nice work!

**Pull request merged**